### PR TITLE
add preferred tab per user for dev perspective Pipelines page and make PipelineRuns tab default for Repository details page

### PR DESCRIPTION
**Pull request by vikram-raj**
**Story:** https://issues.redhat.com/browse/ODC-7229

**GIF:**

Peek 2023-01-11 10-17

Tests:
image

### CODE REVIEW

**Comment by vikram-raj** [comment]
cc: @serenamarie125

**Comment by lokanandaprabhu** [comment]
Tested the functionality. Working as expected. I will add lgtm after test cases are updated.

**Comment by vikram-raj** [comment]
@lokanandaprabhu added e2e tests. PTAL again.

/cc @sanketpathak

**Comment by lokanandaprabhu** [comment]
Hi @vikram-raj ,

One of the e2e test is failing. PTAL.

Screenshot 2023-01-18 at 6 28 17 PM

**Comment by vikram-raj** [comment]
> Hi @vikram-raj ,
> 
> One of the e2e test is failing. PTAL.
> 
> Screenshot 2023-01-18 at 6 28 17 PM

@lokanandaprabhu Fixed it. PTAL

File: frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx [hunk 1]
```diff
@@ -43,7 +46,14 @@ const RepositoryRow: React.FC<RowFunctionArgs<RepositoryKind>> = ({ obj }) => {
   return (
     <>
       <TableData className={repositoriesTableColumnClasses[0]}>
-        <ResourceLink kind={referenceForModel(RepositoryModel)} name={name} namespace={namespace} />
+        <ResourceIcon kind={referenceForModel(RepositoryModel)} />
+        <Link
+          to={`${resourcePath(referenceForModel(RepositoryModel), name, namespace)}/Runs`}
+          className="co-resource-item__resource-name"
+          data-test-id={name}
```
Comment (lokanandaprabhu) [inline frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx:53]: nit: Can we use data-test instead of data-test-id?

**Comment by lokanandaprabhu** [comment]
/lgtm

Requesting @sanketpathak to verify e2e tests.

File: frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx (hunk 1 above)
Comment (vikram-raj) [reply frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx:53]: I added `data-test-id` here as the pre-defined function is looking for `data-test-id`.

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Feature: Perform the actions on Pipelines page
+              As a user, I want to navigate through the Pipeline pages in dev perspective
+
+        Background:
+            Given user has created or selected namespace "aut-pipelines-nav"
+              And user is at pipelines page
+
+        @regression @odc-7131
+        Scenario: Remember last visited tab for the Pipeline page: P-12-TC01
+            Given user is at pipelines page
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature:11]: Already part of background statement

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+             When user navigate to repositories page
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature:12]:
```suggestion
             When user navigates to repositories page
```

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+              And user navigate to builds page
+              And user navigate to pipelines page
+             Then user redirect to repositories page
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature:15]: Can you change the verb in all the following statements
```suggestion
             Then user redirects to repositories page
```

**Comment by sanketpathak** [comment]
@vikram-raj  Is this an expected behavior
Peek 2023-01-20 16-39

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+             Then user will be redirected to Git Repository added page
+              And user clicks on close button
+              And user will be redirected to Repository details page with header name "git-hello-func"
+
+        @regression @odc-7131
+        Scenario: Default tab for respository details page is PipelineRuns if user is not coming from the create repository flow: P-12-TC03
+            Given repository "git-hello-func" is present on the Repositories page
+             When user searches repository "git-hello-func" in repositories page
+              And user clicks repository "git-hello-func" from searched results on Repositories page
+             Then user will be redirects to PipelineRuns tab
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipeline-navigation.feature:32]:
```suggestion
             Then user will be redirected to PipelineRuns tab
```

**Comment by sanketpathak** [comment]
/lgtm
Verified on a cluster using cluster-bot
[Screencast from 01-20-2023 03:53:35 PM.webm](https://user-images.githubusercontent.com/10252230/213701049-7decd911-cbdc-4ba5-9327-acd9a180ff8d.webm)
/label qe-approved

**Comment by vikram-raj** [comment]
Resolved conflict in `frontend/packages/dev-console/integration-tests/support/constants/pageTitle.ts`

@lokanandaprabhu Can you please add `/lgtm` again. Thanks

**Pull request merged**